### ISSUE TITLE
Unable to delete single leaf-list item

### ISSUE THREAD

**Issue by lukasmacko** [CESNET/Netopeer2#15]
When I try to delete a specific leaf-list item
```
<main xmlns="urn:ietf:params:xml:ns:yang:test-module" xmlns:xc="urn:ietf:params:xml:ns:netconf:base:1.0">
        <numbers xc:operation="delete">1</numbers>    
</main>
````
netopeer sends request to delete all instances
```netopeer2-server[23203]: EDIT_CONFIG: leaflist /test-module:main/numbers, operation 4```

**Comment by ZhaoGuangYue** [comment]
After modify ,there is some thing wrong with the xpath when delete all leaf list elements 
as an example ,when use the follow xml to delete the data,the xpath is like 
**/TestDelLeafList:RootContainer/secondLevelLeafList[.='']**
when use this xpath to operator sysrepo data ,it report Error!

`<RootContainer xmlns="urn:ietf:params:xml:ns:yang:fox" xmlns:xc="urn:ietf:params:xml:ns:netconf:base:1.0">
  <secondLevelLeafList xc:operation="delete"></secondLevelLeafList>
</RootContainer>`

**Comment by rkrejci** [comment]
The problem I have found was with numeric values - in case of any (unsigned) int or decimal64 type, the empty string was missinterpret as value 0. This was issue in libyang which should be now fixed (since libyang 0.9.177 and 0.10.52).

I don't know what is the type of your `secondLevelLeafList`, but in case of numeric types, the invalid value `""` should be reported now. In case of e.g. string data type, empty string `""` is in general allowed value, so the sysrepo will be asked to delete the leaf-list with the value `""`. In case the value does not fit the data type restrictions (e.g. length or pattern), it should be reported the same way as in the case of numbers.
